routers/api/v1/api.go

```diff
@@ -664,6 +664,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 				m.Group("/branches", func() {
 					m.Get("", repo.ListBranches)
 					m.Get("/*", context.RepoRefByType(context.RepoRefBranch), repo.GetBranch)
+					m.Delete("/*", context.RepoRefByType(context.RepoRefBranch), repo.DeleteBranch)
```

Comment:
We should require repo (code) writer permissions for this action.

Reply:
how did I missed that